scripts/console/2023.2.0/002_recover_vlans.py

```diff
@@ -79,20 +92,39 @@ def get_range(vlans, avoid) -> list[list[int]]:
                         for val in range(tag_item[0], tag_item[1]+1):
                             in_use_tags[intf_id].add(val)
 
-for intf_id, interface in interfaces.items():
-    vlans = in_use_tags.get(intf_id, set())
-    if OF_LLDP_VLAN:
-        vlans.add(OF_LLDP_VLAN)
-    vlans = get_range(sorted(list(vlans)), set())
-    intf = controller.get_interface_by_id(intf_id)
-    tag_range = intf.tag_ranges["vlan"]
-    available_tags = range_difference(tag_range, vlans)
-    if intf.available_tags["vlan"] != available_tags:
-        print(f"Missing available tags in interface {intf_id}:\n"
-              f"WRONG -> {intf.available_tags['vlan']}\n"
-              f"CORRECT -> {available_tags}")
-        if PRINT_MISSING:
-            print(f"MISSING -> {range_difference(available_tags, intf.available_tags['vlan'])}")
-        print("\n")
-        if not DRY_RUN:
-            intf.make_tags_available(controller, available_tags, 'vlan')
+switch_rm_flows = {}
+for dpid in list(controller.switches.keys()):
+    switch = controller.get_switch_by_dpid(dpid)
+    of_lldp_cookie = get_cookie(switch.id)
+    of_lldp_flow = False
+    for flow in switch.flows.copy():
+        if of_lldp_cookie == flow.cookie:
+            of_lldp_flow = True
+            break
+    if not switch.is_enabled() and of_lldp_flow:
+        print(f"WARNING -> Switch {dpid} is disabled and has LLDP flows.")
+        if REMOVE_LLDP_FLOWS:
+            switch_rm_flows[dpid] = of_lldp_cookie
+
+    for interface in switch.interfaces.copy().values():
+        intf_id = interface.id
+        vlans = in_use_tags.get(intf_id, set())
+        if OF_LLDP_VLAN and switch.is_enabled() and of_lldp_flow:
+            vlans.add(OF_LLDP_VLAN)
+        vlans = get_range(sorted(list(vlans)), set())
+        intf = controller.get_interface_by_id(intf_id)
+        tag_range = intf.tag_ranges["vlan"]
+        available_tags = range_difference(tag_range, vlans)
+        if intf.available_tags["vlan"] != available_tags:
+            print(f"Missing available tags in interface {intf_id}:\n"
+                  f"WRONG -> {intf.available_tags['vlan']}\n"
+                  f"CORRECT -> {available_tags}")
+            if PRINT_MISSING:
+                print(f"MISSING -> {range_difference(available_tags, intf.available_tags['vlan'])}")
+            if not DRY_RUN:
+                intf.make_tags_available(controller, available_tags, 'vlan')
+            print("\n")
+
+# Removing flows if REMOVE_LLDP_FLOWS is True
```

Comment:
If you were to add a conditional `if REMOVE_LLDP_FLOWS and switch_rm_flows:` that would make the intention even clearer while mitigating the need for a comment. I understand it's a bit redundant to readd `REMOVE_LLDP_FLOWS` here but it can improve readability directly in the code as opposed to the comment itself, see what I mean? 

Finally, let's not forget to only call `send_flows` if `DRY_RUN` is False.

Reply:
Added the if statement and added condition `and not DRY_RUN`.